substrate/frame/support/src/traits/try_runtime/decode_entire_state.rs

```diff
@@ -96,37 +109,46 @@ impl core::fmt::Display for TryDecodeEntireStorageError {
 fn decode_storage_info<V: Decode>(
 	info: StorageInfo,
 ) -> Result<usize, Vec<TryDecodeEntireStorageError>> {
-	let mut next_key = info.prefix.clone();
 	let mut decoded = 0;
 
 	let decode_key = |key: &[u8]| match sp_io::storage::get(key) {
 		None => Ok(0),
 		Some(bytes) => {
 			let len = bytes.len();
 			let _ = <V as DecodeAll>::decode_all(&mut bytes.as_ref()).map_err(|_| {
-				vec![TryDecodeEntireStorageError {
+				TryDecodeEntireStorageError {
 					key: key.to_vec(),
 					raw: Some(bytes.to_vec()),
 					info: info.clone(),
-				}]
+				}
 			})?;
 
-			Ok::<usize, Vec<_>>(len)
+			Ok::<usize, _>(len)
 		},
 	};
 
-	decoded += decode_key(&next_key)?;
+	let mut errors = vec![];
+	let mut next_key = Some(info.prefix.clone());
```

Comment:
We should just use `PrefixKeyIterator` here. Not sure why we have written this here "manually" :P

Reply:
Disadvantage of `KeyPrefixIterator` is that it "skips" keys that error. We want to handle the Result type ourselves.